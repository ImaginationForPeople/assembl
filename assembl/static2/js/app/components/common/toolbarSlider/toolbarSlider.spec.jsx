// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import initStoryshots from '@storybook/addon-storyshots';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.3';
/* eslint-enable */

import ToolbarSlider from './toolbarSlider';

initStoryshots({
  storyKindRegex: /^ToolbarSlider$/
});

configure({ adapter: new Adapter() });

describe('<ToolbarSlider /> - with shallow', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ToolbarSlider defaultValue={0} onSliderChange={() => {}} />);
  });

  it('should render a <Slider /> with a cursor', () => {
    expect(wrapper.find('WithStyles(Slider) [thumb]')).toHaveLength(1);
  });
});