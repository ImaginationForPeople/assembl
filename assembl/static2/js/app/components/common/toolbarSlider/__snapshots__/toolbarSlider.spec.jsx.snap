// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Semantic Analysis|ToolbarSlider default 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={50}
      className="MuiSlider-root-1 slider"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStartCapture={[Function]}
      role="slider"
    >
      <div
        className="MuiSlider-container-2"
      >
        <div
          className="MuiSlider-track-3 MuiSlider-trackBefore-4"
          style={
            Object {
              "transform": "translateY(-50%) scaleX(0.5)",
            }
          }
        />
        <div
          className="MuiSlider-thumbWrapper-6"
          style={
            Object {
              "transform": "translateX(50%)",
            }
          }
        >
          <button
            className="MuiButtonBase-root-15 MuiSlider-thumb-7 MuiSlider-thumbIconWrapper-8"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            onTouchStartCapture={[Function]}
            tabIndex="0"
            type="button"
          >
            <div
              className="sliderIcon"
            >
              <p
                className="sliderText"
              >
                50
              </p>
              <img
                alt="Slider thumb icon"
                src="/static2/img/icons/icon-cursor-slider.svg"
              />
            </div>
          </button>
        </div>
        <div
          className="MuiSlider-track-3 MuiSlider-trackAfter-5"
          style={
            Object {
              "transform": "translateY(-50%) scaleX(0.5)",
            }
          }
        />
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      Ã—
    </button>
    <div>
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            Semantic Analysis|ToolbarSlider
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            default
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-r8d96o eyvlbql0"
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#444",
                    }
                  }
                >
                  &lt;
                  ToolbarSlider
                </span>
                <span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      defaultValue
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        {
                        <span
                          style={
                            Object {
                              "color": "#a11",
                            }
                          }
                        >
                          50
                        </span>
                        }
                      </span>
                    </span>
                  </span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      onSliderChange
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        {
                        <span
                          style={
                            Object {
                              "color": "#170",
                            }
                          }
                        >
                          onSliderChange
                        </span>
                        }
                      </span>
                    </span>
                     
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#444",
                    }
                  }
                >
                  /&gt;
                </span>
              </div>
            </div>
            <button
              className="css-1naocv3 eva467m0"
              onClick={[Function]}
            >
              <div
                className="css-lvl6aa eva467m1"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              ToolbarSlider
              " Component
            </h2>
            <table
              className="css-1uhv8nx e1vdo5380"
            >
              <thead>
                <tr>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    property
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    propType
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    required
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    default
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    color
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#22a",
                          "wordBreak": "break-word",
                        }
                      }
                    >
                      #000
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    labelFormatter
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#170",
                        }
                      }
                    >
                      labelFormatter
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    maxValue
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#a11",
                        }
                      }
                    >
                      100
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    minValue
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#a11",
                        }
                      }
                    >
                      0
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Semantic Analysis|ToolbarSlider playground 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax={90}
      aria-valuemin={0}
      aria-valuenow={30}
      className="MuiSlider-root-18 slider"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStartCapture={[Function]}
      role="slider"
    >
      <div
        className="MuiSlider-container-19"
      >
        <div
          className="MuiSlider-track-20 MuiSlider-trackBefore-21"
          style={
            Object {
              "transform": "translateY(-50%) scaleX(0.33333333333333337)",
            }
          }
        />
        <div
          className="MuiSlider-thumbWrapper-23"
          style={
            Object {
              "transform": "translateX(33.333333333333336%)",
            }
          }
        >
          <button
            className="MuiButtonBase-root-32 MuiSlider-thumb-24 MuiSlider-thumbIconWrapper-25"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            onTouchStartCapture={[Function]}
            tabIndex="0"
            type="button"
          >
            <div
              className="sliderIcon"
            >
              <p
                className="sliderText"
              >
                30
              </p>
              <img
                alt="Slider thumb icon"
                src="/static2/img/icons/icon-cursor-slider.svg"
              />
            </div>
          </button>
        </div>
        <div
          className="MuiSlider-track-20 MuiSlider-trackAfter-22"
          style={
            Object {
              "transform": "translateY(-50%) scaleX(0.6666666666666665)",
            }
          }
        />
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      Ã—
    </button>
    <div>
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            Semantic Analysis|ToolbarSlider
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            playground
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-r8d96o eyvlbql0"
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#444",
                    }
                  }
                >
                  &lt;
                  ToolbarSlider
                </span>
                <span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      defaultValue
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        {
                        <span
                          style={
                            Object {
                              "color": "#a11",
                            }
                          }
                        >
                          30
                        </span>
                        }
                      </span>
                    </span>
                  </span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      maxValue
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        {
                        <span
                          style={
                            Object {
                              "color": "#a11",
                            }
                          }
                        >
                          90
                        </span>
                        }
                      </span>
                    </span>
                  </span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      onSliderChange
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        {
                        <span
                          style={
                            Object {
                              "color": "#170",
                            }
                          }
                        >
                          onSliderChange
                        </span>
                        }
                      </span>
                    </span>
                     
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#444",
                    }
                  }
                >
                  /&gt;
                </span>
              </div>
            </div>
            <button
              className="css-1naocv3 eva467m0"
              onClick={[Function]}
            >
              <div
                className="css-lvl6aa eva467m1"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              ToolbarSlider
              " Component
            </h2>
            <table
              className="css-1uhv8nx e1vdo5380"
            >
              <thead>
                <tr>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    property
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    propType
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    required
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    default
                  </th>
                  <th
                    className="css-11b5gui e6fp4ir1"
                  >
                    description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    color
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#22a",
                          "wordBreak": "break-word",
                        }
                      }
                    >
                      #000
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    labelFormatter
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#170",
                        }
                      }
                    >
                      labelFormatter
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    maxValue
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#a11",
                        }
                      }
                    >
                      100
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
                <tr>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    minValue
                  </td>
                  <td
                    className="css-1qcb1f7 e6fp4ir0"
                  >
                    <span>
                      unknown
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    -
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  >
                    <span
                      style={
                        Object {
                          "color": "#a11",
                        }
                      }
                    >
                      0
                    </span>
                  </td>
                  <td
                    className="css-11b5gui e6fp4ir0"
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
